src/Microsoft.Diagnostics.Runtime/src/Linux/LinuxLiveDataReader.cs

```diff
@@ -274,69 +274,87 @@ private void LoadThreads()
         {
-            if (bytesRequested < 1)
+            if (bytesToRead <= 0)
             {
                 return 0;
             }
 
-            ulong endAddress = address + (ulong)bytesRequested - 1;
-            int bytesReadable = 0;
-            ulong prevEndAddr = default;
+            int i = GetMapEntryContaining(address);
+            if (i < 0)
+            {
+                return 0;
+            }
 
-            int startIndex = -1;
-            for (int i = 0; i < _memoryMapEntries.Count; i++)
+            int bytesReadable;
+            ulong prevEndAddr;
             {
                 MemoryMapEntry entry = _memoryMapEntries[i];
-                ulong entryBeginAddr = entry.BeginAddress;
                 ulong entryEndAddr = entry.EndAddress;
-                if (entryBeginAddr <= address && address < entryEndAddr && entry.IsReadable)
-                {
-                    int regionSize = (int)(entryEndAddr - address);
-                    if (regionSize >= bytesRequested)
-                    {
-                        return bytesRequested;
-                    }
 
-                    startIndex = i;
-                    bytesRequested -= regionSize;
-                    bytesReadable = regionSize;
-                    prevEndAddr = entryEndAddr;
-                    break;
+                int regionSize = (int)(entryEndAddr - address);
+                if (regionSize >= bytesToRead)
+                {
+                    return bytesToRead;
                 }
-            }
 
-            if (startIndex < 0)
-            {
-                return 0;
+                bytesToRead -= regionSize;
+                bytesReadable = regionSize;
+                prevEndAddr = entryEndAddr;
             }
 
-            for (int i = startIndex + 1; i < _memoryMapEntries.Count; i++)
+            for (i += 1; i < _memoryMapEntries.Count; i++)
             {
                 MemoryMapEntry entry = _memoryMapEntries[i];
                 ulong entryBeginAddr = entry.BeginAddress;
                 ulong entryEndAddr = entry.EndAddress;
-                if (entryBeginAddr > endAddress || entryBeginAddr != prevEndAddr || !entry.IsReadable)
+                if (entryBeginAddr != prevEndAddr || !entry.IsReadable)
                 {
                     break;
                 }
 
                 int regionSize = (int)(entryEndAddr - entryBeginAddr);
-                if (regionSize >= bytesRequested)
+                if (regionSize >= bytesToRead)
                 {
-                    bytesReadable += bytesRequested;
+                    bytesReadable += bytesToRead;
                     break;
                 }
 
-                bytesRequested -= regionSize;
+                bytesToRead -= regionSize;
                 bytesReadable += regionSize;
                 prevEndAddr = entryEndAddr;
             }
 
             return bytesReadable;
         }
 
+        private int GetMapEntryContaining(ulong address)
```

Comment:
Same question here with `GetRegionContaining`/`GetMapEntryContaining`.  It would be better if we could find a way to merge that code together, maybe by unifying MemoryMapEntry/MemoryRegion...even if that means having to read in one form and convert to another.